en/components/tree.md

```diff
@@ -0,0 +1,313 @@
+
+```html
+<igx-tree>
+	<igx-tree-node *ngFor="let nodes of data" [data]="node" [expanded]="isNodeExpaded(node)" [selected]="isNodeSelected(node)">
+		{{ node.text }}
+		<img [src]="node.image" alt="node.imageAlt" />
+		<igx-tree-node *ngFor="let child of node.children" [data]="child">
+				<a igxTreeNodeLink href="child.url" target="_blank">
+                    {{ child.text }}
+                </a>
+		</igx-tree-node>
+	</igx-tree-node>
+</igx-tree>
+```
+### Node Interactions
+IgxTree provides the following API methods for node interactions:
+- [**expand**]({environment:angularApiUrl}/classes/igxtreenodecomponent.html#expand) - expands the node
+- [**collapse**]({environment:angularApiUrl}/classes/igxtreenodecomponent.html#collapse) - collapses the node
+- [**toggle**]({environment:angularApiUrl}/classes/igxtreenodecomponent.html#toggle) - toggles node expansion state
+- [**collapseAll**]({environment:angularApiUrl}/classes/igxtreecomponent.html#collapseAll) - collapses the specified nodes. If no nodes are passed, collapses all parent nodes.
+- [**expandAll**]({environment:angularApiUrl}/classes/igxtreecomponent.html#expandAll) - sets the specified nodes as expanded. If no nodes are passed, expands all parent nodes.
+- [**deselectAll**]({environment:angularApiUrl}/classes/igxtreecomponent.html#deselectAll) - deselects all nodes. If a nodes array is passed, deselects only the specified nodes. Does not emit nodeSelection event.
+
+### Finding Nodes
+You can find a specific node within IgxTree by using the [findNodes]({environment:angularApiUrl}/classes/igxtreecomponent.html#findNodes) method.  It returns an array of nodes, which match the specified data.
+When finding nodes, you can pass a custom comparer function in order to find the data.
+```html
+<igx-tree>
+	<igx-tree-node *ngFor="let nodes of data" [data]="node" [expanded]="isNodeExpaded(node)" [selected]="isNodeSelected(node)">
+		{{ node.text }}
+		<img [src]="node.image" alt="node.imageAlt" />
+		<igx-tree-node *ngFor="let child of node.children" [data]="child" [expanded]="isNodeExpaded(child)" [selected]="isNodeSelected(child)">
+            {{ child.text }}
+		</igx-tree-node>
+	</igx-tree-node>
+</igx-tree>
+```
+```typescript
+export class MyTreeViewComponent {
+  public data: { [key: string]: any, valueKey: string } = MY_DATA;
+  @ViewChild(IgxTreeComponent, { read: IgxTreeComponent })
+  public tree;
+
+  findNode(valueKey: string) {
+    const comparer: (nodeData: any, node: IgxTreeNodeComponent) => boolean =
+      (data: any, node: IgxTreeNodeComponent) => nodeData.valueKey === data;
+    this.tree.findData(valueKey, comparer);
+  }
+}
+```
+### Templating
+To create a reusable template for your nodes, declare `<ng-template>` **within `igx-tree`**. 
+```html
+ <igx-tree>
+    <igx-tree-node #myNode *ngFor="let node of data" [data]="node">
+        <ng-template *ngTemplateOutlet="nodeTemplate; context { $implicit: myNode }">
```

Comment:
Implementation is incorrect.
With the template being:
```html
<ng-template #nodeTemplate let-node>
        <igx-combo type="line" [(ngModel)]="node.CompanyName" [displayKey]="'CompanyName'" [data]="data" width="220px"></igx-combo>
</ng-template>
```
`node.CompanyName` would be `undefined` when passing `myNode`, as `myNode` is not the data, it's a reference to the `IgxTreeNodeComponent<any>`. The combo would also have no values selected, as its `ngModel` should be bound to an array.

I suggest you go with the following example snippet, so that it's easier for users to unders:
```html
<igx-tree>
    <igx-tree-node *ngFor="let node of data" [data]="node">
        <ng-template *ngTemplateOutlet="#nodeTemplate; context: { $implicit: data }"></ng-template>
        <igx-tree-node *ngFor="let child of node.ChildCompanies" [data]="child">
            <ng-template *ngTemplateOutlet="#nodeTemplate; context: { $implicit: child}"></ng-template>
        </igx-tree-node>
    </igx-tree-node>
    <ng-template #nodeTemplate let-data>
        <div class="node-content-wrapper">
              <igx-icon class="company-logo">{{ data.Logo }}</igx-icon>
              <div class="company-name">{{ data.CompanyName }}</div>
        </div>
    </ng-template>
</igx-tree>
```